yeonjae/programmers-1/과일장수.js

```diff
@@ -0,0 +1,14 @@
+function solution(k, m, score) {
+  const newScore = score.sort((a, b) => a - b);
+  let applePrice = 0;
+
+  while (newScore.length >= m) {
+    const temp = [];
+
+    for (let i = 0; i < m; i++) {
+      temp.push(newScore.pop());
+    }
+    applePrice += Math.min(...temp) * m;
+  }
+  return applePrice;
+}
```

Comment:
```suggestion
function solution(k, m, score) {
  const newScore = [...score].sort((a, b) => a - b);
  let applePrice = 0;

  while (newScore.length >= m) {
    const temp = [];

    for (let i = 0; i < m; i++) {
      temp.push(newScore.pop());
    }
    applePrice += Math.min(...temp) * m;
  }
  return applePrice;
}
```
- sort메서드는 사이드 이펙트가 있기 때문에 원본 배열을 바로 정렬시켜버립니다. 따라서 새로운 변수에 담는 목적이라면 [...score]와 같이 전개연산자로 복사 후 정렬시켜줘야 합니다 🙇‍♂️